### ISSUE TITLE
LogBuilder from Fluent throws exception when used wrongly

### ISSUE THREAD

**Issue by dlidstrom** [NLog/NLog#4067]
**NLog version**: 4.7.2

**Platform**: .Net 4.0

The fluent API can be misused, and in that case it throws an exception. For example, I (wrongly) assumed that I should put the log text immediately in Info:

```csharp
Log.Info(jsonText);
```

I didn't read up on the documentation properly. But, this now failed with an exception:

```txt
System.ArgumentException: Illegal characters in path.
   at System.IO.Path.CheckInvalidPathChars(String path, Boolean checkAdditional)
   at System.IO.Path.GetFileName(String path)
   at System.IO.Path.GetFileNameWithoutExtension(String path)
   at NLog.Fluent.Log.Create(LogLevel logLevel, String callerFilePath)
```

I'm coming from log4net where exceptions are never thrown by the logging framework. This lack of error handling takes away quite a bit from NLog, unfortunately (IMO of course).
I can think of a simple error handling in this case. If the callerFilePath can't be parsed, use a default or omit entirely. But never throw an exception. Please 🙇‍♂️

**Comment by snakefoot** [comment]
Yes it is a strange pattern to check for filenames on the disk when doing logging. Never understood the need for this.

Hoping to replace Fluent-API with something better in NLog 5.0 (#4044). But yes exceptions should not be thrown, even when calling methods that does silly things like checking for filenames on the disk.

**Comment by snakefoot** [comment]
Note that NLog do have some input validation. Ex. these calls will throw:

```c#
LogManager.GetLogger(null);
LogManager.GetLogger("Test").Log((LogLevel)null, "Hello");
```

**Comment by snakefoot** [comment]
Examples of NLog Fluent can be found here:

https://github.com/NLog/NLog/wiki/Fluent-API

**Comment by dlidstrom** [comment]
Yes I did find the docs when I realised it wasn't doing at all what I thought. I actually thought the global Log instance would replace all my local, static readonly instances, with no syntax differences in the actual logging calls. That was great, I thought.

**Comment by snakefoot** [comment]
Resolved with #4069
